examples/flutter_gallery/lib/demo/material/bottom_app_bar_demo.dart

```diff
@@ -13,291 +13,311 @@ class BottomAppBarDemo extends StatefulWidget {
+    title: 'Free - End',
+    label: 'floating action button floats above the end of the bottom app bar',
+    value: FloatingActionButtonLocation.endFloat,
+  );
+
+  static const _ChoiceValue<FloatingActionButtonLocation> kFabCenterFloat = const _ChoiceValue<FloatingActionButtonLocation>(
+    title: 'Free - Center',
+    label: 'floating action button is floats above the center of the bottom app bar',
+    value: FloatingActionButtonLocation.centerFloat,
+  );
 
-  static const List<_FabLocationConfiguration> _fabLocationConfigurations = const <_FabLocationConfiguration>[
-    const _FabLocationConfiguration('End, undocked above the bottom app bar', _BabMode.END_FAB, FloatingActionButtonLocation.endFloat),
-    const _FabLocationConfiguration('End, docked to the bottom app bar', _BabMode.END_FAB, FloatingActionButtonLocation.endDocked),
-    const _FabLocationConfiguration('Center, docked to the bottom app bar', _BabMode.CENTER_FAB, FloatingActionButtonLocation.centerDocked),
-    const _FabLocationConfiguration('Center, undocked above the bottom app bar', _BabMode.CENTER_FAB, FloatingActionButtonLocation.centerFloat),
-    // This configuration uses a custom FloatingActionButtonLocation.
-    const _FabLocationConfiguration('Start, docked to the top app bar', _BabMode.CENTER_FAB, const _StartTopFloatingActionButtonLocation()),
-  ];
-  
-  // The index of the currently-selected _FabShapeConfiguration.
-  int fabShapeIndex = 1;
-
-  static const List<_FabShapeConfiguration> _fabShapeConfigurations = const <_FabShapeConfiguration>[
-      const _FabShapeConfiguration('None', null),
-      const _FabShapeConfiguration('Circular', 
-        const FloatingActionButton(
-          onPressed: _showSnackbar,
-          child: const Icon(Icons.add),
-          backgroundColor: Colors.orange,
-        ),
-      ),
-      const _FabShapeConfiguration('Diamond',
-        const _DiamondFab(
-          onPressed: _showSnackbar,
-          child: const Icon(Icons.add),
-        ),
-      ),
-    ];
+  static void _showSnackbar() {
+    const String text =
+      "When the Scaffold's floating action button location changes, "
+      'the floating action button animates to its new position.'
+      'The BottomAppBar adapts its shape appropriately.';
+    _scaffoldKey.currentState.showSnackBar(
+      const SnackBar(content: const Text(text)),
+    );
+  }
+
+  // App bar color
 
-  // The currently-selected Color for the Bottom App Bar.
-  Color babColor;
-
-  // Accessible names for the colors that a Screen Reader can use to
-  // identify them.
-  static final Map<Color, String> colorToName = <Color, String> {
-    null: 'White',
-    Colors.orange: 'Orange',
-    Colors.green: 'Green',
-    Colors.lightBlue: 'Light blue',
-  };
-  static const List<Color> babColors = const <Color> [
+  static const List<Color> kBabColors = const <Color>[
```

Comment:
Kebabs anyone?  Since this is demo code, maybe use something more descriptive than "Bab", since it might not be obvious that "bab" is the bottom app bar.